### PR TITLE
Frontier/natspec

### PR DESCRIPTION
**Pull request by zsfelfoldi**
NatSpec, end-to-end test, cli option --natspec

### CODE REVIEW

**Comment by obscuren** [comment]
Please rebase

**Comment by obscuren** [comment]
Please update the following:

* Change back the **genesis block**
* Rebase again
* Pick an address that will be used for the NatSpec

File: xeth/xeth.go [hunk 1]
```diff
@@ -608,6 +608,12 @@ func (self *XEth) Call(fromStr, toStr, valueStr, gasStr, gasPriceStr, dataStr st
 	return common.ToHex(res), err
 }
 
+func (self *XEth) ConfirmTransaction(tx string) bool {
```
Comment (obscuren) [inline xeth/xeth.go]: Why is transactions a string?

Comment (zsfelfoldi) [inline xeth/xeth.go]: Because natspec.js needs the whole JSON-encoded RPC request, so it was easier to pass it than to re-encode.

**Comment by fjl** [comment]
It looks like the tests for `common/natspec` are still hanging on Travis.

**Comment by zsfelfoldi** [comment]
Sure, in its final form URL reg should be one single call with a variable length parameter. Tried that with no success, even checked the EVM byte code which made no sense to me. Maybe the compiler can't handle this yet?

**Pull request merged**